pkg/kindsys/load.go

```diff
@@ -0,0 +1,234 @@
+func (decl *SomeDecl) IsCustomStructured() bool {
+	_, is := decl.Meta.(CustomStructuredMeta)
+	return is
+}
+
+// IsComposable indicates whether the represented kind is a composable kind.
+func (decl *SomeDecl) IsComposable() bool {
+	_, is := decl.Meta.(ComposableMeta)
+	return is
+}
+
+// Decl represents a single kind declaration, having been loaded
+// and validated by a func such as [LoadCoreKind].
+//
+// Its type parameter indicates the category of kind.
+type Decl[T KindMetas] struct {
+	// V is the cue.Value containing the entire Kind declaration.
+	V cue.Value
+	// Meta contains the kind's metadata settings.
+	Meta T
+}
+
+// Some converts the typed Decl to the equivalent typeless SomeDecl.
+func (decl *Decl[T]) Some() *SomeDecl {
+	return &SomeDecl{
+		V:    decl.V,
+		Meta: any(decl.Meta).(SomeKindMeta),
+	}
+}
+
+// LoadCoreKind loads and validates a core kind declaration of the kind category
+// indicated by the type parameter. On success, it returns a [Decl] which
+// contains the entire contents of the kind declaration.
+//
+// declpath is the path to the directory containing the core kind declaration,
+// relative to the grafana/grafana root. For example, dashboards are in
+// "kinds/structured/dashboard".
+//
+// The .cue file bytes containing the core kind declaration will be retrieved
+// from the central embedded FS, [grafana.CueSchemaFS]. If desired (e.g. for
+// testing), an optional fs.FS may be provided via the overlay parameter, which
+// will be merged over [grafana.CueSchemaFS]. But in typical circumstances,
+// overlay can and should be nil.
+//
+// This is a low-level function, primarily intended for use in code generation.
+// For representations of core kinds that are useful in Go programs at runtime,
+// see ["github.com/grafana/grafana/pkg/registry/corekind"].
+func LoadCoreKind[T RawMeta | CoreStructuredMeta](declpath string, ctx *cue.Context, overlay fs.FS) (*Decl[T], error) {
+	vk, err := cuectx.BuildGrafanaInstance(declpath, "kind", ctx, overlay)
+	if err != nil {
+		return nil, err
+	}
+	decl := &Decl[T]{
+		V: vk,
+	}
+	decl.Meta, err = ToKindMeta[T](vk)
+	if err != nil {
+		return nil, err
+	}
+	return decl, nil
+}
+
+// LoadCustomKind takes an fs.FS that contains a CustomKind declaration at the
+// path indicated by dir, and validates that it contains a valid CustomKind
+// declaration. On success, it returns a Go representation of the entire
+// CustomKind declaration.
+func LoadCustomKind(kfs fs.FS, dir string, ctx *cue.Context) (*Decl[CustomStructuredMeta], error) {
+	inst, err := cuectx.LoadInstanceWithGrafana(kfs, dir)
+	panic("TODO")
+}
+
+// NOTE exists solely for discussion purposes
+func LoadCustomKind2(kfs fs.FS, dir string, ctx *cue.Context) (*CustomKind, error) {
+	panic("JUST SPITBALLING")
+}
+
+// NOTE exists solely for discussion purposes
+type CustomKind struct {
+	Decl    *Decl[CustomStructuredMeta]
+	Lineage thema.Lineage
+}
```

Comment:
@IfSentient i added these temporarily solely to provoke some conversation with you - `LoadCustomKind` and `LoadCustomKind2` here are both forms i could easily imagine adding, if they're something you could reasonably layer on top of. Won't be this PR, but i know you're thinking about this right now.

`CustomKind` would be filled out to become a generic version of what we're generating per-kind under `pkg/kinds/*/*_kind_gen.go`

 of these forms preferable? Or are